### DISCUSSION TITLE
Documentation + Scraping

### DISCUSSION THREAD

**Discussion by SMCaseyCode** [masecla22/Reddit4J#7] [Q&A] [answered]
Hello! I am pretty new when it comes to API usage. I am currently making a Discord bot and have it working, but wanting to scrape Reddit and have it output posts from a specific subreddit. I have been trying to get this API wrapper to work, but I am having major difficulties. I have been mainly trying to use .getSubredditPosts but only ever get outputs like "masecla.reddit4j.requests.SubredditPostListingEndpointRequest@c267ef4". I honestly have no idea how to make use of this wrapper and do what I need to do. The documentation seems very barebones and I can't find anything on google using this wrapper to help. All I can ever find is Python usages!!! Is there anywhere where there is extra documentation besides the wiki?

**Comment by masecla22** [comment]
Can drop a code example?

**Reply by SMCaseyCode** [reply]
Of course, I would like to preface this again by saying I am VERY new to API's (so much so, this is my first time using one besides the basic JDA discord bot one). So far, I have just thrown statements around trying to get an idea of what they do, hence some of the mess and obviously very little implementation. I'm sure I am missing a TON of core code, just not sure what I can reference to understand what I need. I also just used println as an effort to get any kind of idea of what's being output. I have tried a ton of methods on the lines seen here, this is just what I ended with tonight before calling it a night.

```
    public Reddit4J getClient() throws AuthenticationException, IOException, InterruptedException {

        Reddit4J client = Reddit4J.rateLimited().setUsername("REDACTED").setPassword("REDACTED")
                .setClientId("REDACTED").setClientSecret("REDACTED")
                .setUserAgent(new UserAgentBuilder().appname("REDACTED").author("REDACTED").version("1.0"));

        client.connect();

            System.out.println(client.getSubredditPosts("JavaBotTestNet", Sorting.NEW )); //this outputs what I mentioned in last comment.

            System.out.println(client.getSubreddit("JavaBotTestNet").getNew()); //If I remove ".getNew()" I get what I expect out.

        //"JavaBotTestNet" is just a random subreddit I made specifically to test this.


        return client;
    }
```

I would also like to add that Maven gives the error "Dependency 'com.github.masecla22:Reddit4J:master-SNAPSHOT' not found" for this code (C+P off of this github)
```
        <dependency>
            <groupId>com.github.masecla22</groupId>
            <artifactId>Reddit4J</artifactId>
            <version>master-SNAPSHOT</version>
        </dependency>
```

**Comment by SMCaseyCode** [comment, marked as answer]
Update: I ended up just using OkHttp and Jackson to do everything.

EDIT: At the time of the original post I was very new to Java. I rewrote the program I was working on and I solved my original issue rather quickly. I needed to add .submit() to the end of the getSubredditPosts method call. Works great now! My "solution" beforehand worked, but with my lack of Java understanding, it was messy and inefficient and laced with bugs.
